### ISSUE TITLE
Run all tests from this file

### ISSUE THREAD

**Issue by simonvpe** [fredericbonnet/cmake-test-explorer#19]
is not working, the test sources don't seem to be linked to the test cases picked up by the plugin

**Comment by fredericbonnet** [comment]
Hi, and thanks for using my extension!

I can't find the command "Run all tests from this file", do you mean "Test Explorer: Run tests in current file"?

Anyway, CMake doesn't support linking source files to a test case, so this command can't be supported with the CMake test adapter.

**Comment by simonvpe** [comment]
It is a nice extension and I'm evaluating it for our project. Yes that is
probably what I meant. What is required to get the source file to test
link? Could it be achieved with some glue code somewhere?

**Comment by fredericbonnet** [comment]
When I started working on this extension I made some experiments by settings custom CTest properties but alas this gave no results, as it only keeps the properties it knows. But on second though I think it should be achievable with carefully chosen environment variables using the `ENVIRONMENT` test property. For example:

```
add_test(
    NAME <name>
    COMMAND <command> [<args> ...]
)
set_tests_properties(<name> PROPERTIES
    ENVIRONMENT "TEST_FILE=<filename>;TEST_LINE=<line>"
)
```

Then when the extension calls `ctest --show-only=json-v1` to build the test list, it can get the variables like so:

```json
      "name" : "<name>",
      "properties" : 
      [
        {
          "name" : "ENVIRONMENT",
          "value" : 
          [
            "TEST_FILE=<filename>",
            "TEST_LINE=<line>"
          ]
        },
        {
          "name" : "WORKING_DIRECTORY",
          "value" : "<dir>"
        }
      ]
```

The only caveat is that you have to choose environment variables that don't collide with anything, and of course the extension needs extra config settings. But it should work and would improve the developer experience greatly. I can add this feature very easily if you're ready to experiment with it.

**Comment by simonvpe** [comment]
That sounds like something that will work for us. I would appreciate it
very much if this was added :)

**Comment by fredericbonnet** [comment]
I've just published a new version 0.11.0 that adds the feature (all the relevant info is in the README). Feedback welcome ;)

**Comment by fredericbonnet** [comment]
BTW if your projects are plain C (not C++) and you're looking for a fast, simple & easy to integrate testing framework, I've authored [PicoTest](https://github.com/fredericbonnet/picotest), a single-file unit testing framework for C. I have a (still unpublished) VSCode extension similar to this one that provides a great developer experience and very fast runtimes.
